src/kademlia/node.cpp

```diff
@@ -101,22 +102,23 @@ void incoming_error(entry& e, char const* msg, int error_code = 203)
 
 } // anonymous namespace
 
-node::node(udp proto, udp_socket_interface* sock
+node::node(aux::session_listen_socket* sock, socket_manager* sock_man
```

Comment:
In the case of the DHT the problem is we use the pointer as a key into a `std::map` in `dht_tracker` so we can't store it purely as a reference. Although I guess we could use a reference wherever possible and add ref/deref operators as necessary.